src/Libraries/Microsoft.Extensions.AI.Abstractions/ChatCompletion/ChatResponseUpdate.cs

```diff
@@ -38,6 +38,29 @@ public class ChatResponseUpdate
     /// <summary>The name of the author of the update.</summary>
     private string? _authorName;
 
+    /// <summary>Initializes a new instance of the <see cref="ChatResponseUpdate"/> class.</summary>
+    [JsonConstructor]
+    public ChatResponseUpdate()
+    {
+    }
+
+    /// <summary>Initializes a new instance of the <see cref="ChatResponseUpdate"/> class.</summary>
+    /// <param name="role">The role of the author of the update.</param>
+    /// <param name="contents">The text contents of the update.</param>
+    public ChatResponseUpdate(ChatRole? role, string? contents)
```

Comment:
Since this reflects a single `TextContent` i would keep the singular name for the property.

```suggestion
    public ChatResponseUpdate(ChatRole? role, string? textContent)
```

```suggestion
    public ChatResponseUpdate(ChatRole? role, string? content)
```

Reply:
I went back and forth on this one. But since you raise it, I can change it back.